gateway/readconfig.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+// OsEnv implements interface to wrap os.Getenv
+type OsEnv struct {
+}
+
+// Getenv wraps os.Getenv
+func (OsEnv) Getenv(key string) string {
+	return os.Getenv(key)
+}
+
+// HasEnv provides interface for os.Getenv
+type HasEnv interface {
+	Getenv(key string) string
+}
+
+// ReadConfig constitutes config from env variables
+type ReadConfig struct {
+}
+
+func parseBoolValue(val string) bool {
+	if val == "true" {
+		return true
+	}
+	return false
+}
+
+func parseIntValue(val string) int {
+	if len(val) > 0 {
+		parsedVal, parseErr := strconv.Atoi(val)
+
+		if parseErr == nil && parsedVal >= 0 {
+
+			return parsedVal
+		}
+	}
+	return 0
+}
+
+// Read fetches config from environmental variables.
+func (ReadConfig) Read(hasEnv HasEnv) GatewayConfig {
+	cfg := GatewayConfig{
+		httpPort: 8080,
+	}
+
+	readTimeout := parseIntValue(hasEnv.Getenv("read_timeout"))
+	writeTimeout := parseIntValue(hasEnv.Getenv("write_timeout"))
+
+	if readTimeout == 0 {
+		readTimeout = 8
+	}
+
+	if writeTimeout == 0 {
+		writeTimeout = 8
+	}
+
+	cfg.readTimeout = time.Duration(readTimeout) * time.Second
+	cfg.writeTimeout = time.Duration(writeTimeout) * time.Second
+
+	return cfg
+}
+
+// GatewayConfig for the process.
+type GatewayConfig struct {
+	httpPort     int
```

Comment:
This was still hard-coded -- just moved it into the config struct.
I have removed it since it was out of scope.